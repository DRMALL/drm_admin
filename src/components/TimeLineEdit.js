


import React from 'react'
import getInputValue from '../actions/getInputValue'
import { recordKind } from '../commons/machineLable'
import Select from 'antd/lib/select'
import Input from 'antd/lib/input'
import selectValue from '../utils/selectValue'
import DatePicker from 'antd/lib/date-picker'
import dispatch from '../actions/dispatch'
import moment from 'moment'



export default props => {
  const { line_time, line_type, line_des } = props
  return (
    <div  >
      <div className='machine-input-flex' >
        <div className='machine-input-left' >时间</div>
        <DatePicker onChange={ getTime } style={{ width: 400 }}
                    placeholder={ moment(line_time).format('YYYY-MM-DD') } />
      </div>
      <div className='machine-input-flex' >
          <div  className='machine-input-left'>分类</div>
          <Select style={{ width: 400 }} placeholder='选择分类'
                  onChange={ selectValue } defaultValue={ line_type } >
                  {
                    recordKind.map((item, index) =>
                      <Select.Option key={ index } value={ item+'/line_type' } >{ item }</Select.Option>)
                  }
          </Select>
        </div>
        <div className='machine-input-flexs' >
        <div className='machine-input-left' >描述</div>
        <Input.TextArea rows={4} className='machine-input-right'
                  onChange={ getInputValue }
                  style={{ width: 400 }} placeholder='输入描述'
                  data-path='MACHINE' data-id='line_des' defaultValue={ line_des } />
      </div>
    </div>
    )
}

function getTime(ment){
  dispatch('MACHINE_SELCET_TIME', ment)
}


