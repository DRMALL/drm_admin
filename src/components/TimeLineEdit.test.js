

import React from 'react'
import { render } from 'enzyme'
import toJson from 'enzyme-to-json'
import TimeLineEdit from './TimeLineEdit'

it('TimeLineEdit test', ()=> {
  const line_type = 'line_type'
  const line_des = 'line_des'
  const wrapper = render(<TimeLineEdit line_type={ line_type } line_des={ line_des } />)
  expect(toJson(wrapper)).toMatchSnapshot()
})
