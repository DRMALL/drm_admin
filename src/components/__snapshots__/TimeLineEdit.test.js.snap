// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeLineEdit test 1`] = `
<div>
  <div
    class="machine-input-flex"
  >
    <div
      class="machine-input-left"
    >
      时间
    </div>
    <span
      class="ant-calendar-picker"
      style="width:400px;"
    >
      <div>
        <input
          class="ant-calendar-picker-input ant-input"
          placeholder="2017-08-08"
          readonly=""
          value=""
        />
        <span
          class="ant-calendar-picker-icon"
        />
      </div>
    </span>
  </div>
  <div
    class="machine-input-flex"
  >
    <div
      class="machine-input-left"
    >
      分类
    </div>
    <div
      class="ant-select ant-select-enabled"
      style="width:400px;"
    >
      <div
        aria-autocomplete="list"
        aria-expanded="false"
        aria-haspopup="true"
        class="ant-select-selection
                  ant-select-selection--single"
        role="combobox"
        tabindex="0"
      >
        <div
          class="ant-select-selection__rendered"
        >
          <div
            class="ant-select-selection__placeholder"
            style="display:none;user-select:none;-webkit-user-select:none;"
            unselectable="unselectable"
          >
            选择分类
          </div>
          <div
            class="ant-select-selection-selected-value"
            style="display:block;opacity:1;"
            title="line_type"
          >
            line_type
          </div>
        </div>
        <span
          class="ant-select-arrow"
          style="user-select:none;-webkit-user-select:none;"
          unselectable="unselectable"
        >
          <b />
        </span>
      </div>
    </div>
  </div>
  <div
    class="machine-input-flexs"
  >
    <div
      class="machine-input-left"
    >
      描述
    </div>
    <textarea
      class="ant-input machine-input-right"
      data-id="line_des"
      data-path="MACHINE"
      placeholder="输入描述"
      rows="4"
      style="width:400px;"
    >
      line_des
    </textarea>
  </div>
</div>
`;
